pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -563,33 +552,60 @@ func getGpuDriverVersion(vmSize string) string {
 // IsSgxEnabledSKU determines if an VM SKU has SGX driver support.
 func getIsSgxEnabledSKU(vmSize string) bool {
 	switch vmSize {
-	case "Standard_DC2s", "Standard_DC4s":
+	case vmSizeStandardDc2s, vmSizeStandardDc4s:
 		return true
 	}
 	return false
 }
 
 func getShouldConfigureHTTPProxy(httpProxyConfig *nbcontractv1.HTTPProxyConfig) bool {
+	if httpProxyConfig == nil {
+		return false
+	}
 	return httpProxyConfig.GetHttpProxy() != "" || httpProxyConfig.GetHttpsProxy() != ""
 }
 
 func getShouldConfigureHTTPProxyCA(httpProxyConfig *nbcontractv1.HTTPProxyConfig) bool {
+	if httpProxyConfig == nil {
```

Comment:
same as above, getter will return empty string which will make the condition false